apps/browser/lib/services/evo/EvoThread.ts

```diff
@@ -159,22 +159,13 @@ export class EvoThread {
     const chatId = this._config.chatId;
     this._state.isLoading = true;
 
-    const results = await Promise.all<[
-      Promise<ChatLog[]>,
-      Promise<Workspace>
-    ]>([
-      this._config.loadChatLog(chatId).catch((reason) => {
-        this._callbacks?.onError(reason.toString());
-        return [];
-      }),
-      this._config.loadWorkspace(chatId).catch((reason) => {
-        this._callbacks?.onError(reason.toString());
-        return new InMemoryWorkspace();
-      })
-    ]);
-
-    this._state.logs = results[0];
-    this._state.workspace = results[1];
+    const logs = await this._config.loadChatLog(chatId).catch((reason) => {
+      this._callbacks?.onError(reason.toString());
+      return [];
+    });
+
+    this._state.logs = logs;
+    this._state.workspace = this._config.loadWorkspace(chatId);
```

Comment:
Should probably figure out how we want to best handle error-states within the EvoThread. Currently it's a bit undefined...

Ideally the UI would halt it's use of the EvoService + EvoThread if it enters into an unrecoverable situation (ex: unable to load chat history or workspace given a chat ID).

Reply:
This is ofc for another issue / PR. Curious to hear your thoughts on this though.

Reply:
I don't have any immediate thoughts on this
We should create an issue and tackle it whenever

Reply:
Issue: https://github.com/polywrap/evo.ninja/issues/637